x-pack/plugins/apm/public/components/shared/Links/apm/AnomalyDetectionSetupLink.tsx

```diff
@@ -28,7 +27,7 @@ export function AnomalyDetectionSetupLink() {
     (callApmApi) =>
       callApmApi({ pathname: `/api/apm/settings/anomaly-detection` }),
     [],
-    { preservePreviousData: false }
+    { preservePreviousData: false, showToastOnError: false }
```

Comment:
The backend now throws an error when ML is not available. We should simply ignore that on the client in this case.
The better solution in this case is to check the permissions and license on the client first, and only make the request if both those are ok.